app/controllers/shared/api.php

```diff
@@ -49,44 +50,164 @@
+                $queueForUsage
+                    ->addMetric("functions", $value); // per project
+
+                if ($event === Database::EVENT_DOCUMENT_DELETE) {
+                    //Project level function deduction
+                    $deployments = $dbForProject->getDocument('stats', md5("_inf_function." . "{$document->getId()}" . ".deployments"));
+                    $deploymentsStorage  = $dbForProject->getDocument('stats', md5("_inf_function." . "{$document->getId()}" . ".deployments.storage"));
+                    $builds  = $dbForProject->getDocument('stats', md5("_inf_" . "{$document->getId()}" . ".builds"));
+                    $buildsStorage = $dbForProject->getDocument('stats', md5("_inf_" . "{$document->getId()}" . ".builds.storage"));
+                    $buildsCompute = $dbForProject->getDocument('stats', md5("_inf_" . "{$document->getId()}" . ".builds.compute"));
+                    $executions = $dbForProject->getDocument('stats', md5("_inf_" . "{$document->getId()}" . ".executions"));
+                    $executionsCompute = $dbForProject->getDocument('stats', md5("_inf_" . "{$document->getId()}" . ".executions.compute"));
+
+                    $queueForUsage
+                        ->addMetric("deployments", ($deployments['value'] * -1))
+                        ->addMetric("deployments.storage", ($deploymentsStorage['value'] * -1))
+                        ->addMetric("builds", ($builds['value'] * -1))
+                        ->addMetric("builds.storage", ($buildsStorage['value'] * -1))
+                        ->addMetric("builds.compute", ($buildsCompute['value'] * -1))
+                        ->addMetric("executions", ($executions['value'] * -1))
+                        ->addMetric("executions.compute", ($executionsCompute['value'] * -1))
+                    ;
+                }
+                break;
+            case $document->getCollection() === 'deployments':
+                $queueForUsage
+                    ->addMetric("deployments", $value) // per project
+                    ->addMetric("deployments.storage", $document->getAttribute('size') * $value) // per project
+                    ->addMetric("{$document['resourceType']}" . "." . "{$document['resourceId']}" . ".deployments", $value)// per function
+                    ->addMetric("{$document['resourceType']}" . "." . "{$document['resourceId']}" . ".deployments.storage", $document->getAttribute('size') * $value) // per function
+                ;
+                break;
+            case $document->getCollection() === 'executions':
+                $queueForUsage
+                    ->addMetric("executions", $value) // per project
+                    ->addMetric("{$document['functionId']}" . ".executions", $value) // per function
+                ;
+                break;
+            default:
+                break;
+        }
+    } catch (Throwable $error) {
+        if (!empty($logger) && $error->getCode() === 500) {
```

Comment:
what if error code is not `500` ? again that will fail the API?

Reply:
ok